ocean_lib/assets/asset.py

```diff
@@ -258,19 +258,14 @@ def remove_publisher_trusted_algorithm_publisher(
 
     @enforce_types
     def generate_trusted_algorithms(self) -> dict:
-        algo_metadata = self.metadata
+        """Returns a trustedAlgorithm dictionary for service at index 0."""
+        files = self.get_service_by_index(0).files
```

Comment:
I think this would be better on the service itself. 0 is not guaranteed to be the correct service.

Reply:
Done.

Reply:
I ended up reverting this change due to circular import: Service <--> Asset